### ISSUE TITLE
SSL error on start-up

### ISSUE THREAD

**Issue by ishaqx** [bitnami/bitnami-docker-wordpress#295]
Hi, firstly thank you for creating a great docker image for WP.

On docker-compose start-up I get the following error message: 
wordpress_1  | [Wed May 26 13:26:13.902182 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name

I think this is becuase of the default httpd.conf. Is there a reason why the default configuration was left activated - since it's not (as far as I can see)  needed for the WP instance, which runs of it's own vhost config?

Is there a way to remove this configuraiton without manually editing the httpd.conf file and restarting the apache service? 

Many thanks for your help.

**Comment by alemorcuq** [comment]
The image has HTTPS enabled by default. Could you be more specific about what you mean with the issue in the default `httpd.conf`? I don't see any mentions to port 8443 in that file.

**Comment by ishaqx** [comment]
Hi, thanks for your response.

The error message is happening because the SSL certs I'm using are for a different domain that I'm using for WordPress.

In the httpd.conf there is a domain that's preconfigered called example.com, and since apache can't match the SSL cert for it, it's throwing that error message. 

Since a domain/site for WordPress is configured via its own vhosts conf file, the configuration for example.com in httpd.conf is surpurflous and not needed. That preconfigered example site/domain makes since for an apache only container, but not for a WordPress container. 

You can view the example.com site after firing up the docker container and creating host entry for example.com and pointing it to 127.0.0.1 and then connecting to the specified port on that container from your host machine. 

The bottom line is configuration for example.com should should not be in the httpd.conf file, unless I'm missing something and there's a valid reason why it's included for the bitnami WordPress docker image. 

Many thanks for your help.

**Comment by ishaqx** [comment]
Hi sorry couldn't post details of the set-up earlier on, but I've posted them below for your reference. You are right, the httpd.conf doesn't listen on port 8443, however, I think the issue is being caused by the "ServerAlias *" directive in the wordpress-https-vhost.conf file. I have tried to set ServerAlias to the domain being used for wordpress so that the cert for it is only used for that domain, but it gets overwritten when docker-compose up is issued. However the ServerAlias directive does not get overwritten in the wordpress-vhost.conf file.


root@rna:/persistent-docker-cons/mywordpress# ls -l
total 36

drwxr-xr-x 3 root root 4096 May 27 21:25 back-ups
drwxr-xr-x 2 1001 1001 4096 May 26 14:08 certs
-rw-r--r-- 1 root root 1418 May 27 21:13 docker-compose.yml
drwxr-xr-x 2 1001 1001 4096 May 26 13:24 otherdomain
drwxr-xr-x 3 1001 1001 4096 May 26 13:33 mariadb
drwxr-xr-x 3 1001 1001 4096 May 27 21:09 vhosts
drwxr-xr-x 3 1001 1001 4096 May 26 13:37 wordpress


root@rna:/persistent-docker-cons/mywordpress# cat docker-compose.yml
version: '2'
services:
  mariadb:
    image: docker.io/bitnami/mariadb:10.3
    volumes:
      - /persistent-docker-cons/mywordpress/mariadb:/bitnami/mariadb
    environment:
      - ALLOW_EMPTY_PASSWORD=no
      - MARIADB_USER=bn_wordpress
      - MARIADB_DATABASE=bitnami_wordpress
      - MARIADB_ROOT_PASSWORD=********
      - MARIADB_PASSWORD=********
  wordpress:
    image: docker.io/bitnami/wordpress:5
    ports:
      - '80:8080'
      - '443:8443'
    volumes:
      - /persistent-docker-cons/mywordpress/wordpress:/bitnami/wordpress
      - /persistent-docker-cons/mywordpress/certs:/certs:ro
      - /persistent-docker-cons/mywordpress/vhosts/:/opt/bitnami/apache/conf/vhosts/
      - /persistent-docker-cons/mywordpress/otherdomain:/otherdomain
    depends_on:
      - mariadb
    environment:
      - WORDPRESS_ENABLE_MULTISITE=yes
      - WORDPRESS_MULTISITE_NETWORK_TYPE=subdirectory
      - WORDPRESS_MULTISITE_HOST=wordpress.domain
      - ALLOW_EMPTY_PASSWORD=no
      - WORDPRESS_DATABASE_HOST=mariadb
      - WORDPRESS_DATABASE_PORT_NUMBER=3306
      - WORDPRESS_DATABASE_USER=bn_wordpress
      - WORDPRESS_DATABASE_PASSWORD=********
      - WORDPRESS_DATABASE_NAME=bitnami_wordpress
      - WORDPRESS_USERNAME=********
      - WORDPRESS_PASSWORD=********
      - WORDPRESS_EMAIL=m@wordpress.domain
      - WORDPRESS_FIRST_NAME=********
      - WORDPRESS_LAST_NAME=********
      - WORDPRESS_BLOG_NAME=SOME_NAME



root@rna:/persistent-docker-cons/mywordpress# ls -l certs
total 32
-rw-r--r-- 1 1001 1001 1826 May 26 14:07 otherdomain.pem
-rw------- 1 1001 1001 1704 May 26 14:07 otherdomainpriv.pem
lrwxrwxrwx 1 1001 1001   13 May 26 12:57 server.crt -> wordpress.domain.pem
lrwxrwxrwx 1 1001 1001   13 May 26 12:57 server.key -> wordpress.domainpriv.pem
-rw-r--r-- 1 1001 1001 1838 May 26 12:57 wordpress.domain.crt
-rw------- 1 1001 1001 1704 May 26 12:57 wordpress.domain.key


root@rna:/persistent-docker-cons/mywordpress# ls -la vhosts/
total 28
drwxr-xr-x 3 1001 1001 4096 May 27 21:09 .
drwxr-xr-x 8 root root 4096 May 27 21:25 ..
-rw-r--r-- 1 1001 1001  163 May 27 13:33 00_status-vhost.conf
drwxr-xr-x 2 1001 1001 4096 May 27 13:37 htaccess
-rw-r--r-- 1 1001 1001  631 May 26 14:24 otherdomain.conf
-rw-r--r-- 1 1001 1001 1514 May 27 21:21 wordpress-https-vhost.conf
-rw-r--r-- 1 root root 1426 May 27 21:09 wordpress-vhost.conf


root@rna:/persistent-docker-cons/mywordpress# cat vhosts/*

*** 00_status-vhost.conf ***

<VirtualHost 127.0.0.1:8080>
  ServerName status.localhost
  <Location /server-status>
    Require local
    SetHandler server-status
  </Location>
</VirtualHost>

**** otherdomain.conf **** 

<VirtualHost *:8080>
  ServerName otherdomain.dom
  Redirect permanent / https://otherdomain.com
</VirtualHost>

<VirtualHost *:8443>
  ServerName otherdomain.dom
  SSLEngine on
  SSLCertificateFile "/opt/bitnami/apache/conf/bitnami/certs/otherdomain.pem"
  SSLCertificateKeyFile "/opt/bitnami/apache/conf/bitnami/certs/otherdomainpriv.pem"
  DocumentRoot "/otherdomain"
  <Directory "/otherdomain">
    Options Indexes FollowSymLinks
    AllowOverride All
    Require all granted
  </Directory>
</VirtualHost>

****wordpress-vhost.conf ****

<VirtualHost 127.0.0.1:8080 _default_:8080>
  ServerAlias wordpress.domain
Redirect permanent / https://wordpress.domain/
  Include "/opt/bitnami/apache/conf/vhosts/htaccess/wordpress-htaccess.conf"
</VirtualHost>

**** wordpress-https-vhost.conf ****

<VirtualHost 127.0.0.1:8443 _default_:8443>
  ServerAlias *
  SSLEngine on
  SSLCertificateFile "/opt/bitnami/apache/conf/bitnami/certs/server.crt"
  SSLCertificateKeyFile "/opt/bitnami/apache/conf/bitnami/certs/server.key"
  DocumentRoot /opt/bitnami/wordpress
  <Directory "/opt/bitnami/wordpress">
    Options -Indexes +FollowSymLinks -MultiViews
    AllowOverride None
    Require all granted
    # BEGIN WordPress fix for plugins and themes
    # Certain WordPress plugins and themes do not properly link to PHP files because of symbolic links
    # https://github.com/bitnami/bitnami-docker-wordpress-nginx/issues/43
    RewriteEngine On
    RewriteRule ^bitnami/wordpress(/.*) $1 [L]
    # END WordPress fix for plugins and themes
    # BEGIN WordPress Multisite
    # Using subfolder network type: https://wordpress.org/support/article/htaccess/#multisite
    RewriteEngine On
    RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
    RewriteBase /
    RewriteRule ^index\.php$ - [L]
    # add a trailing slash to /wp-admin
    RewriteRule ^([_0-9a-zA-Z-]+/)?wp-admin$ $1wp-admin/ [R=301,L]
    RewriteCond %{REQUEST_FILENAME} -f [OR]
    RewriteCond %{REQUEST_FILENAME} -d
    RewriteRule ^ - [L]
    RewriteRule ^([_0-9a-zA-Z-]+/)?(wp-(content|admin|includes).*) $2 [L]
    RewriteRule ^([_0-9a-zA-Z-]+/)?(.*\.php)$ $2 [L]
    RewriteRule . index.php [L]
    # END WordPress Multisite
  </Directory>
  Include "/opt/bitnami/apache/conf/vhosts/htaccess/wordpress-htaccess.conf"
</VirtualHost>


*** Logged information when docker-compose is started ***

root@rna:/persistent-docker-cons/mywordpress# docker-compose up
Starting mywordpress_mariadb_1 ... done
Starting mywordpress_wordpress_1 ... done
Attaching to mywordpress_mariadb_1, mywordpress_wordpress_1
mariadb_1    | mariadb 20:56:20.53
mariadb_1    | mariadb 20:56:20.54 Welcome to the Bitnami mariadb container
mariadb_1    | mariadb 20:56:20.54 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb
mariadb_1    | mariadb 20:56:20.56 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb/issues
mariadb_1    | mariadb 20:56:20.56
mariadb_1    | mariadb 20:56:20.57 INFO  ==> ** Starting MariaDB setup **
mariadb_1    | mariadb 20:56:20.64 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1    | mariadb 20:56:20.65 INFO  ==> Initializing mariadb database
mariadb_1    | mariadb 20:56:20.70 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb_1    | mariadb 20:56:20.71 INFO  ==> Setting user option
mariadb_1    | mariadb 20:56:20.74 INFO  ==> Using persisted data
wordpress_1  | wordpress 20:56:21.90
mariadb_1    | mariadb 20:56:20.86 INFO  ==> Running mysql_upgrade
wordpress_1  | wordpress 20:56:21.91 Welcome to the Bitnami wordpress container
mariadb_1    | mariadb 20:56:20.87 INFO  ==> Starting mariadb in background
wordpress_1  | wordpress 20:56:21.91 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-wordpress
wordpress_1  | wordpress 20:56:21.92 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-wordpress/issues
wordpress_1  | wordpress 20:56:21.92
wordpress_1  | wordpress 20:56:21.93 INFO  ==> ** Starting WordPress setup **
wordpress_1  | wordpress 20:56:21.98 INFO  ==> Mounting certificates files from '/certs'
wordpress_1  | wordpress 20:56:22.01 INFO  ==> Configuring PHP options
wordpress_1  | wordpress 20:56:22.05 INFO  ==> Validating settings in MYSQL_CLIENT_* env vars
wordpress_1  | wordpress 20:56:22.12 WARN  ==> The WordPress configuration file '/opt/bitnami/wordpress/wp-config.php' is not writable. Configurations based on environment variables will not be applied for this file.
wordpress_1  | wordpress 20:56:22.34 WARN  ==> The wordpress virtual host file '/opt/bitnami/apache/conf/vhosts/wordpress-vhost.conf' is not writable. Configurations based on environment variables will not be applied for this file.
wordpress_1  | wordpress 20:56:22.36 INFO  ==> Restoring persisted WordPress installation
wordpress_1  | wordpress 20:56:22.40 INFO  ==> Trying to connect to the database server
mariadb_1    | mariadb 20:56:23.59 INFO  ==> Stopping mariadb
mariadb_1    | mariadb 20:56:29.27 INFO  ==> ** MariaDB setup finished! **
mariadb_1    |
mariadb_1    | mariadb 20:56:29.31 INFO  ==> ** Starting MariaDB **
mariadb_1    | 2021-05-27 20:56:29 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.3.29-MariaDB) starting as process 1 ...
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Using Linux native AIO
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Uses event mutexes
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Number of pools: 1
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Using SSE2 crc32 instructions
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: 128 out of 128 rollback segments are active.
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Creating shared tablespace for temporary tables
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Waiting for purge to start
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: 10.3.29 started; log sequence number 8325467; transaction id 4822
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
mariadb_1    | 2021-05-27 20:56:29 0 [Note] InnoDB: Buffer pool(s) load completed at 210527 20:56:29
mariadb_1    | 2021-05-27 20:56:29 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1    | 2021-05-27 20:56:29 0 [Note] Server socket created on IP: '0.0.0.0'.
mariadb_1    | 2021-05-27 20:56:29 0 [Warning] 'proxies_priv' entry '@% root@e0f45721842e' ignored in --skip-name-resolve mode.
mariadb_1    | 2021-05-27 20:56:29 0 [Note] Reading of all Master_info entries succeeded
mariadb_1    | 2021-05-27 20:56:29 0 [Note] Added new Master_info '' to hash table
mariadb_1    | 2021-05-27 20:56:29 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
mariadb_1    | Version: '10.3.29-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
wordpress_1  | wordpress 20:56:35.00 INFO  ==> ** WordPress setup finished! **
wordpress_1  |
wordpress_1  | wordpress 20:56:35.01 INFO  ==> ** Starting Apache **
wordpress_1  | [Thu May 27 20:56:35.085451 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
wordpress_1  | [Thu May 27 20:56:35.086970 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
wordpress_1  | [Thu May 27 20:56:35.126361 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
wordpress_1  | [Thu May 27 20:56:35.127576 2021] [ssl:warn] [pid 1] AH01909: www.example.com:8443:0 server certificate does NOT include an ID which matches the server name
wordpress_1  | [Thu May 27 20:56:35.160303 2021] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.46 (Unix) OpenSSL/1.1.1d PHP/7.4.19 configured -- resuming normal operations
wordpress_1  | [Thu May 27 20:56:35.164029 2021] [core:notice] [pid 1] AH00094: Command line: '/opt/bitnami/apache/bin/httpd -f /opt/bitnami/apache/conf/httpd.conf -D FOREGROUND'

Please let me know if I can provide any other information.

**Comment by alemorcuq** [comment]
I'm going to create an internal task to investigate this. In any case, this is only a warning and your installation will work well despite it showing.

**Comment by ishaqx** [comment]
Great, thanks!

**Comment by carrodher** [comment]
In order to unify the approaches followed in Bitnami containers and Bitnami charts, we are moving some issues in `bitnami/bitnami-docker-<container>` repositories to `bitnami/containers`. Please follow [bitnami/containers](https://github.com/bitnami/containers) to keep you updated about the latest bitnami images.

If this issue is still applicable, please feel free to recreate it at https://github.com/bitnami/containers/issues/new/choose

More information here: https://blog.bitnami.com/2022/07/new-source-of-truth-bitnami-containers.html
